From 677716c2471cadf1bf1268eca6343498a886a229 Mon Sep 17 00:00:00 2001
From: Mikhail Ryzhov <mikhail.ryzhov@intel.com>
Date: Wed, 10 Sep 2025 15:41:31 +0200
Subject: [PATCH] [CMAKE] Fixed version detection with cmake 4.1 (#32039)

### Details:
- Starting from CMake 4.0, version variables are set to "" if not
defined by project(). See policy CMP0048.
 - if(${var}) will check that variable is not empty too

### Tickets:
 - *ticket-id*

Upstream-Status: Backport
[https://github.com/openvinotoolkit/openvino/commit/677716c2471cadf1bf1268eca6343498a886a229]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 cmake/developer_package/version.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/developer_package/version.cmake b/cmake/developer_package/version.cmake
index 0718931298..98cd4a4ffe 100644
--- a/cmake/developer_package/version.cmake
+++ b/cmake/developer_package/version.cmake
@@ -99,7 +99,8 @@ macro(ov_parse_ci_build_number repo_root)
         endforeach()
 
         foreach(var OpenVINO_VERSION_MAJOR OpenVINO_VERSION_MINOR OpenVINO_VERSION_PATCH)
-            if(DEFINED ${var} AND NOT ${var} EQUAL ${var}_HPP)
+            # Starting from CMake 4.0, version variables are set to "" if not defined by project(). See policy CMP0048.
+            if(${var} AND NOT ${var} EQUAL ${var}_HPP)
                 message(FATAL_ERROR "${var} parsed from CI_BUILD_NUMBER (${${var}}) \
                     and from openvino/core/version.hpp (${${var}_HPP}) are different")
             else()
-- 
2.34.1

